src/flint/fq_default.jl

```diff
@@ -436,43 +430,19 @@ end
 #
 ###############################################################################
 
-function *(x::Int, y::FqFieldElem)
-  z = parent(y)()
-  ccall((:fq_default_mul_si, libflint), Nothing,
-        (Ref{FqFieldElem}, Ref{FqFieldElem}, Int, Ref{FqField}), z, y, x, y.parent)
-  return z
-end
-
-*(x::Integer, y::FqFieldElem) = ZZRingElem(x)*y
-
-*(x::FqFieldElem, y::Integer) = y*x
-
-function *(x::ZZRingElem, y::FqFieldElem)
-  z = parent(y)()
-  ccall((:fq_default_mul_fmpz, libflint), Nothing,
-        (Ref{FqFieldElem}, Ref{FqFieldElem}, Ref{ZZRingElem}, Ref{FqField}),
-        z, y, x, y.parent)
-  return z
+for jT in (Integer, ZZRingElem)
+  @eval begin
+    *(x::FqFieldElem, y::$jT) = mul!(parent(x)(), x, y)
+    *(x::$jT, y::FqFieldElem) = mul!(parent(y)(), x, y)
+    
+    +(x::FqFieldElem, y::$jT) = x + parent(x)(y)
+    +(x::$jT, y::FqFieldElem) = y + x
+    
+    -(x::FqFieldElem, y::$jT) = x - parent(x)(y)
+    -(x::$jT, y::FqFieldElem) = parent(y)(x) - y
```

Comment:
shouldn't these delegate to `add!` or `sub!` as well?
Edit: I just noticed that there are no corresponding flint functions for these, but we have fallbacks in place in AA.
Edit: but this would be an endless loop...

maybe just add a comment here that these wait for a change in FLINT?

Reply:
We could implement `add!` and `sub!` already now without an endless loop by inserting the coercions there. But there are already tons of changes and I have to draw the line somewhere. So here I just took the code that was there, the only change is that it is now in an `for ... @begin eval` construct.

Someone can make these additional changes in a future iteration. There are a gazillion other things one could do in this file alone (e.g. adding `set!` methods  to simplify the constructors). For now I'd like to move on